src/typing/typer.ml

```diff
@@ -1597,6 +1623,19 @@ and type_call ?(mode=MGet) ctx e el (with_type:WithType.t) inline p =
 		(match follow e.etype with
 			| TFun signature -> type_bind ctx e signature args p
 			| _ -> def ())
+	| (EField (e,"start"),_), args ->
```

Comment:
What are your thoughts on expressing this in the type-system instead? This reminds me of the `trace` problem where we have a variable amount of arguments followed by `PosInfos`, except here we have the `_hx_continuation` argument instead.

Reply:
I haven't thought about this. I wonder if TypeScript "mapped types" feature has something that could allow this, but I don't know to what extent that makes sense in Haxe. Also @Aurel300 is our type system expert, maybe he has some ideas?

We could also have compiler-supported callable abstracts for these specific cases, like `CoroutineStart<(CoroArg1,CoroArg2)->CoroReturn>` (and similar for `.bind` and `trace`), but I'm not sure if that'a good idea either...